lib/offline/download_manager.js

```diff
@@ -130,6 +138,22 @@ shaka.offline.DownloadManager = class {
     return this.estimator_.getTotalDownloaded();
   }
 
+  updateInitData_(initData) {
+    const normalisedInitData = shaka.util.Pssh.normaliseInitData(initData, true);
+    this.drmEngine_.newInitData('cenc', normalisedInitData);
+    for (const period of this.manifest_.periods) {
+      for (const variant of period.variants) {
+        for (const drmInfos of variant.drmInfos) {
+          drmInfos.initData = [
+            {initDataType: 'cenc', initData: normalisedInitData}
+          ];
+        }
+      }
+    }
+    const variants = shaka.util.Periods.getAllVariantsFrom(this.manifest_.periods);
+    this.drmEngine_.init_(variants);
```

Comment:
This feels very hacky but i have to call this to update the `currentDrmInfo_` object. Is there a better way to update this?

Reply:
I think that re-initializing the DRM engine without the storage module knowing, so to speak, is a bit risky design-wise. So I agree that this is hacky.

It feels like you could move this stuff out of the download manager and into storage.js. Like, do the checks to see if it contains a pssh and such in there, and then if appropriate update the manifest and re-initialize the DRM engine. The storage module already filters the manifest, so that feels like the "right" place to put this stuff. It would also mean you would no longer have to pass in the manifest and such to the download manager.

As far as how to actually re-initialize the DRM engine, you should try calling `DrmEngine.initForStorage` again. It's kind of a thin layer over `DrmEngine.init_`, without being a private method. If that doesn't work, you should make a new method in the DRM engine for it, that fits this specific use-case.